### ISSUE TITLE
config list with -c flag for the not extended file

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#18]
I've tested `config list` both with and without `-c` flag and got the same output.
Would it make sense to implement the `-c` flag for the basic file as well?

**Comment by buhtignew** [comment]
I think I've got the answer already while testing `config show` command:
the basic file doesn't have categories, but only labels.

**Comment by buhtignew** [comment]
Maybe we should implement `config list -c` only for extended file, i.e. without the need of `-e` flag, this the description of the `-c` flag in help would be `-c, --categories: Shows list of available categories in extended file.`

**Comment by d5000** [comment]
`-c` is normally meant to be one of these "optional positional arguments" (i.e. normally you can simply enter `pacli config list CATEGORY` without `-c`).

But in this case, your idea probably makes sense because you anyway have to enter a flag to access the extended config file.

I would however, in this case, implement it in the form: `pacli config list -e CATEGORY` to preserve the "extended" flag/terminology used in the other config commands, and to highlight that the user is dealing here with the extended config file.

The same treatment could also be implemented in `config set` and `config show`, i.e. the `-c` flag would be gone completely in this group of commands.

Thinking about it I think it doesn't have any drawbacks so I'll implement it in the next days. Please leave this issue open until then :)

**Comment by buhtignew** [comment]
In such a case we can implement the `-c` flag for the basic file as well. For the moment it would output only one category, but in case the PeerAssets team goes on developing their project again adding other categories for the basic file we'll be easily able follow them.
As alternative, we can output the message for the `config list -c` that there are no categories for the basic file. 
My goal here is to reduce the info to be read and understood in the help file, because often people do not dedicate enough time to it.
I don't know whether it makes that much sense though.

**Comment by d5000** [comment]
The idea with the a message when `-c` is used without `-e` is good imo. I've implemented it for the next commit.

> I think we can't suppress the -c flag because a concise list of categories is very useful for extended file.
These are two different `-c` flags. ( I was first also confused yesterday, but the `-c` in `config show/set` was almost never needed, so I removed it).

> What I have discovered is that one can get the same list just putting any word before the -e flag

True. This is however not a bug, because `config list` only accepts boolean flags (keyword arguments) and no positional arguments, so any words entered as positional arguments are simply ignored.
